openmdao/docs/usr-guide/tutorials/file-passing.rst

```diff
@@ -0,0 +1,231 @@
+File Passing Tutorial
+======================
+
+References to files can be passed between Components in OpenMDAO using variables
+called `FileRefs`.  A `FileRef` is just an object that contains a file name and
+an absolute file path which is calculated by the framework. The *fname* attribute
+can be a simple name or a name that includes a relative or absolute
+directory path.
+
+Calculation of Absolute Directory Paths
+---------------------------------------
+
+During setup, OpenMDAO determines the absolute file system path for each
+`FileRef` variable based on the directory path of the `Component`
+that contains it.  The process works like this:
+
+1) Starting at the root `System` of the tree, we calculate its absolute directory
+   based on its 'directory' attribute.  If the 'directory' attribute is empty,
+   then the absolute directory is just the current working directory. If
+   *directory* contains a relative pathname, then the absolute directory is
+   the current working directory plus the relative path.  If *directory* is
+   already an absolute path, then we just use that.  If *directory* is a
```

Comment:
Move this discussion about the function way of specifying in to a note:: block. Add. "Sometimes you don't want to hard-code the directory name. If you want to delay picking a name until runtime, you can use specify `directory` as a function..."